test/types.ts

```diff
@@ -0,0 +1,5 @@
+import { expect } from 'tstyche'
+
+export function expectType<T>(expression: T) {
+  return expect(expression).type.toBe<T>()
```

Comment:
Are you sure this works? Types only statically analysed, so this would pass because type of `expression` is `T`. All other checks would only check assignability in `expectType<T>(expression: T)` level without reaching this line.